orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/DelegatingExecutionRepository.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipeline.persistence
+
+/**
+ * Allows writing delegating execution repository extensions, while continuing
+ * to expose the concrete internal implementation.
+ */
+interface DelegatingExecutionRepository<T : ExecutionRepository> : ExecutionRepository {
+  fun getDelegate(): T
+}
```

Comment:
This isn't used by OSS yet, but is necessary for `-nflx`'s `DualExecutionRepository` implementation. That will be pulled forward later.